src/v4r_util/rviz_visualizer.py

```diff
@@ -0,0 +1,109 @@
+
+import rospy
+from visualization_msgs.msg import MarkerArray, Marker
+from v4r_util.util import  o3d_bb_to_ros_bb
+from vision_msgs.msg import BoundingBox3DArray
+
+class RvizVisualizer:
+    '''
+    Allows for easy visualization of BoundingBox3DArray in rviz.
+    Initialize with a topic name, and then order the visualization via
+    namespace. Cam publish ros and o3d bounding boxes, 
+    aswell as clear markers from certain namespaces.
+    '''
+
+    def __init__(self, topic="TablePlaneExtractorVisualizer"):
+        self.pub = rospy.Publisher(
+            topic, 
+            MarkerArray, 
+            queue_size=10)
+        self.markers = {}
+
+    def publish_ros_bb(self, ros_bb, namespace="", clear_old_markers=True):
+        '''
+        Publishes a single BoundingBox3D to rviz in the given namespace.
+        '''
+        marker_arr = self.ros_bb_arr_to_rviz_marker_arr([ros_bb], namespace, clear_old_markers)
+        self.pub.publish(marker_arr)
+
+    def publish_ros_bb_arr(self, ros_bb_arr, namespace="", clear_old_markers=True):
+        '''
+        Publishes a BoundingBox3DArray to rviz in the given namespace.
+        '''
+        marker_arr = self.ros_bb_arr_to_rviz_marker_arr(ros_bb_arr, namespace, clear_old_markers)
+        self.pub.publish(marker_arr)
+
+
+    def publish_o3d_bb_arr(self, o3d_bb_arr, header, namespace="", clear_old_markers=True):
+        '''
+        Publishes a o3d boundig box arr to rviz in the given namespace.
+        '''
+
+        if o3d_bb_arr is None:
+            print("No bounding boxes found")
+            return
+        ros_bb_arr = BoundingBox3DArray()
+        ros_bb_arr.header = header
+        ros_bb_arr.boxes = [o3d_bb_to_ros_bb(bb) for bb in o3d_bb_arr]
+        marker_arr = self.ros_bb_arr_to_rviz_marker_arr(ros_bb_arr, namespace, clear_old_markers)
+        self.pub.publish(marker_arr)
+
+    def ros_bb_to_rviz_marker(self, ros_bb, namespace="", id=0, header=None):
+        '''
+        Converts BoundingBox3D to rviz Marker.
+        Input: vision_msgs/BoundingBox3D ros_bb
+        Output: visualization_msgs/Marker marker
+        '''
+        marker = Marker()
+        marker.header.frame_id = header.frame_id
```

Comment:
Header should not be an optional argument with None as a default value if it is always needed.